unpack_action.go

```diff
@@ -12,12 +12,23 @@ type UnpackAction struct {
 	File        string
 }
 
+func UnpackTarArchive(infile, destination, compression string) error {
+
+	unpackTarOpt := map[string]string{
+		"gz":    "z",
+		"bzip2": "j",
+		"xz":    "J",
+	} // Trying to guess all other supported formats
+
+	os.MkdirAll(destination, 0755)
+	log.Printf("Unpacking %s\n", infile)
+
+	return Command{}.Run("unpack", "tar", "-x"+unpackTarOpt[compression], "-f", infile, "-C", destination)
```

Comment:
not sure about it.
Modern tar implementations can guess compression type if you do not provide any option.
btw, this is the reason why there is no 'default' for compression switch.